pkg/sync/http/http_sync.go

```diff
@@ -151,18 +153,16 @@ func (hs *Sync) generateSha(body []byte) string {
 
 func (hs *Sync) Fetch(ctx context.Context) (string, error) {
 	if hs.URI == "" {
-		hs.ready = false
 		return "", errors.New("no HTTP URL string set")
 	}
 
 	body, err := hs.fetchBodyFromURL(ctx, hs.URI)
 	if err != nil {
-		hs.ready = false
 		return "", err
 	}
 	if len(body) != 0 {
 		hs.LastBodySHA = hs.generateSha(body)
 	}
-	hs.ready = true
+
 	return string(body), nil
 }
```

Comment:
I don't know how I missed this before :facepalm:

Reply:
I also did not think about the implications of the ready state when we introduced the probe.